aiven/resource_flink_job.go

```diff
@@ -0,0 +1,233 @@
+	client := m.(*aiven.Client)
+
+	project := d.Get("project").(string)
+	serviceName := d.Get("service_name").(string)
+	jobName := d.Get("job_name").(string)
+	jobStatement := d.Get("statement").(string)
+	jobTables := flattenToString(d.Get("tables").([]interface{}))
+
+	createRequest := aiven.CreateFlinkJobRequest{
+		JobName:   jobName,
+		Statement: jobStatement,
+		Tables:    jobTables,
+	}
+
+	createResponse, err := client.FlinkJobs.Create(project, serviceName, createRequest)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	jobId := createResponse.JobId
+
+	conf := &resource.StateChangeConf{
+		Pending: []string{
+			"CANCELED",
+			"CANCELING",
+			"CREATED",
+			"DEPLOYING",
+			"FAILED",
+			"FINISHED",
+			"INITIALIZING",
+			"RECONCILING",
+			"SCHEDULED",
+		},
+		Target: []string{
+			"RUNNING",
+		},
+		Refresh: func() (interface{}, string, error) {
+			r, err := client.FlinkJobs.Get(project, serviceName, aiven.GetFlinkJobRequest{JobId: jobId})
+			if err != nil {
+				return nil, "", err
+			}
+			return r, r.State, nil
+		},
+		Delay:      1 * time.Second,
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 1 * time.Second,
+	}
+
+	r, err := conf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("Error waiting for job to become active: %s", err)
+	}
+	getResponse := r.(*aiven.GetFlinkJobResponse)
+
+	if err := d.Set("project", project); err != nil {
+		return diag.Errorf("error setting Flink Jobs `project` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("service_name", serviceName); err != nil {
+		return diag.Errorf("error setting Flink Jobs `project` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("job_id", getResponse.JID); err != nil {
+		return diag.Errorf("error setting Flink Jobs `job_id` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("job_name", getResponse.Name); err != nil {
+		return diag.Errorf("error setting Flink Jobs `job_name` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("state", getResponse.State); err != nil {
+		return diag.Errorf("error setting Flink Jobs `state` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("statement", jobStatement); err != nil {
+		return diag.Errorf("error setting Flink Jobs `statement` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("tables", jobTables); err != nil {
+		return diag.Errorf("error setting Flink Jobs `tables` for resource %s: %s", d.Id(), err)
+	}
+	d.SetId(buildResourceID(project, serviceName, getResponse.JID))
+
+	return resourceFlinkJobRead(ctx, d, m)
+}
+
+func resourceFlinkJobDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	project := d.Get("project").(string)
+	serviceName := d.Get("service_name").(string)
+	jobId := d.Get("job_id").(string)
+
+	// TODO: the flink proxy api terminates job with a PATCH request; we use that as a "delete" notion
```

Comment:
Let's remove this TODO from the code. Since we do not have a `delete` API ATM, we can track it in JIRA or GitHub issues.

Reply:
:heavy_check_mark: